scripts/utils/docs-content-tools/handlers.js

```diff
@@ -0,0 +1,142 @@
+};
+
+/**
+ * Reads a file and separates the content and frontmatter data
+ * @param {String} path - The full filepath of the file
+ * @returns {Object} Original path, slug from path, content, frontmatter data
+ */
+const separateData = async (path) => {
+  const file = fs.readFileSync(path, 'utf8');
+  const slug = getSlugFromPath(path);
+  const separateData = await frontmatterGH(file);
+  const { data: frontmatterData, content } = separateData;
+  if (!frontmatterData.redirects) {
+    frontmatterData.redirects = [];
+  }
+  return { path, content, frontmatterData, slug };
+};
+
+/**
+ * Joins the content and frontmatter data
+ * @param {Object} data - content, frontmatterData, path
+ * @returns {Object} Combined data and path
+ */
+const joinData = ({ content, frontmatterData, path }) => {
+  const data = frontmatterGH.stringify(content, frontmatterData, {
+    lineWidth: -1,
+  });
+
+  return { path, data };
+};
+
+/**
+ * Adds a redirect if the slug is not already in the redirects array
+ * @param {Object} data - slug, frontmatterData, rest
+ * @returns {Object} Edited frontmatterData, whether the redirect was added or skipped
+ */
+const addRedirect = ({ frontmatterData, slug, ...rest }) => {
+  if (frontmatterData.redirects.includes(slug)) {
+    return { frontmatterData, ...rest, skipped: slug };
+  }
+  frontmatterData.redirects.push(slug);
+  return {
+    frontmatterData,
+    ...rest,
+    added: slug,
+  };
+};
+
+/**
+ * Removes a redirect if the slug is in the redirects array
+ * @param {Object} data - slug, frontmatterData, rest
+ * @returns {Object} Edited frontmatterData, whether the redirect was removed or skipped
+ */
+const removeRedirect = ({ frontmatterData, slug, ...rest }) => {
+  if (frontmatterData.redirects.includes(slug)) {
+    frontmatterData.redirects = frontmatterData.redirects.filter(
+      (path) => path !== slug
+    );
+    if (frontmatterData.redirects.length === 0) {
+      delete frontmatterData.redirects;
+    }
+    return {
+      frontmatterData,
+      ...rest,
+      removed: slug,
+    };
+  }
+  return { frontmatterData, ...rest, skipped: slug };
+};
+
+/**
+ * Globs an array of all mdx files in a directory + sub-directories
+ * @param {String} directory - The parent directory to search
+ * @returns {String[]} All mdx files in the directory and sub-directories
+ */
+const getFilePaths = (directory) => {
+  if (directory.length === 0) {
+    // if user did not supply paths, default to all
+    return glob.sync(`${__dirname}/../../../src/content/docs/**/*.mdx`);
+  } else {
+    return glob.sync(`${__dirname}/../../../src/content${directory}**/*.mdx`);
+  }
+};
+
+/**
+ * Retrieves the user inputs from command line
+ * @returns {Object} The directory and action specified by the user
+ */
+const getUserInputs = () => {
+  const userInputs = process.argv.slice(2);
+  const actionIndex = userInputs.findIndex(
+    (arg) => arg === '--add' || arg === '--remove'
+  );
```

Comment:
This might be a good opportunity to reach for [`Array.prototype.some`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/some) which returns true if at least one element passes the conditional statement:

```js
const hasAction = userInputs.some((input) => ['--add', '--remove'].includes(input));

if (!hasAction) {
  // complain to the user
}
```

Reply:
Ah, but it looks like you probably want to find the first action...well then that's probably a case for [`Array.prototype.find`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/find) instead. Sorry I should have read down a few lines first.

```js
const actionInput = userInputs.find((input) => ['--add', '--remove'].includes(input));

if (!actionInput) {
  // complain to the user
}

const action = actionInput.replace('--', '');
```

Personally, I think that reads a little more clearly too "if there's no action input, complain". A little easier to follow than the `actionIndex === -1` stuff.

Reply:
yep